packages/common/src/core/schemas/internal/fetchCategoriesUiSchemaElement.ts

```diff
@@ -1,40 +1,130 @@
- * @param i18nScope i18n scope for the entity translations
- * @param entity The entity to build the UI schema for
- * @returns the UI schema element for thumbnail editing
+ * @param opts Options for fetching categories UI schema element.
+ * @param opts.source The source of categories, either "org" or "query".
+ * @param opts.currentValues Current values to consider when building the UI schema element.
+ * @param opts.context The ArcGIS context containing request options and portal information.
+ * @param opts.query The query to use if the source is "query".
+ * @returns An array of UI schema elements for categories, including recognized and unrecognized categories.
  */
 export async function fetchCategoriesUiSchemaElement(
-  i18nScope: string,
-  context: IArcGISContext
+  opts: IFetchCategoriesUiSchemaElementOptions
 ): Promise<IUiSchemaElement[]> {
-  const categoryItems = await fetchCategoryItems(
-    context.portal.id,
-    context.hubRequestOptions
+  const recognizedCategories =
+    opts.source === "org"
+      ? await fetchOrgCategories(
+          opts.context.portal.id,
+          opts.context.hubRequestOptions
+        )
+      : await fetchQueryCategories(opts.query, opts.context.hubRequestOptions);
+
+  const unrecognizedValues = (opts.currentValues || []).filter(
+    (value) => !recognizedCategories.includes(value)
   );
 
+  // Split unrecognized values into full and partial categories
+  const unrecognizedFullCategories: string[] = [];
+  const unrecognizedPartialCategories: string[] = [];
+  unrecognizedValues.forEach((value) => {
+    isFullyQualifiedCategory(value)
+      ? unrecognizedFullCategories.push(value)
+      : unrecognizedPartialCategories.push(value);
+  });
+
+  // Convert the recognized categories to nested combobox items
+  const recognizedItems = toNestedComboboxItems(recognizedCategories);
+
+  // Convert the unrecognized fully-qualified categories to nested combobox items
+  const unrecognizedFullItems = toNestedComboboxItems(
+    unrecognizedFullCategories
+  );
+
+  // Convert the unrecognized partial categories to top-level combobox items since we don't have a full path for them
+  const unrecognizedPartialItems: IUiSchemaComboboxItem[] =
+    unrecognizedPartialCategories.map((category) => ({
+      label: category,
+      value: category,
+      children: [] as IUiSchemaComboboxItem[],
+    }));
+
+  // Place partial items at the front for better visibility
+  const unrecognizedItems = [
+    ...unrecognizedPartialItems,
+    ...unrecognizedFullItems,
+  ];
+
+  const fieldOptions: Record<string, unknown> = {
+    control: "hub-field-input-combobox",
+    allowCustomValues: false,
+    selectionMode: "ancestors",
+    placeholderIcon: "select-category",
+  };
+
+  // Construct groups for the combobox items
+  fieldOptions.groups = [
+    !!unrecognizedItems.length && {
+      // TODO: Fix label key
+      label:
+        "{{shared.fields.categories.unrecognizedCategoriesGroup.label:translate}}",
```

Comment:
This label is still being worked on my Justin, I'll update once he decides the final value